File: RoRServer/src/main/java/commands/editor/StartDirectGameCommand.java
package commands.editor;

import java.util.UUID;

import commands.base.CommandBase;
import communication.MessageInformation;
import models.game.Player;
import models.session.EditorSession;
import models.session.EditorSessionManager;
import models.session.RoRSession;

public class StartDirectGameCommand extends CommandBase {

	private UUID playerId;
	private boolean isHost;
	public StartDirectGameCommand(RoRSession session, MessageInformation messageInfo) {
		super(session, messageInfo);
		playerId = messageInfo.getValueAsUUID("playerId");
		isHost = messageInfo.getValueAsBoolean("isHost");
	}

	@Override
	public void execute() {
		Player player = session.getPlayerById(playerId);

		// Alle Spieler löschen wenn Spieler Host ist
		if (isHost) {
			EditorSessionManager.getInstance().removeEditorSession((EditorSession) session, true);
		} 
	}

}

File: RoRClient/Commands/Editor/Other/StartDirectGameCommand.cs
﻿using RoRClient.Commands.Base;
using RoRClient.Communication.DataTransferObject;
using RoRClient.Models.Game;
using RoRClient.Models.Session;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RoRClient.Commands.Editor.Other
{
    class StartDirectGameCommand : CommandBase
    {
        public StartDirectGameCommand(RoRSession session, MessageInformation message) : base(session, message)
        {

        }
        public override void Execute()
        {
            EditorSession.GetInstance().DeleteEditorSession();
            GameSession.GetInstance().Started = true;
        }
    }
}
